indexing-service/src/main/java/io/druid/indexing/common/task/Task.java

```diff
@@ -128,6 +135,20 @@
   public boolean isReady(TaskActionClient taskActionClient) throws Exception;
 
   /**
+   * Returns whether or not this task can restore its progress from its on-disk working directory. Restorable tasks
+   * may be started with a non-empty working directory. Tasks that exit uncleanly may still have a chance to attempt
+   * restores, meaning that restorable tasks should be able to deal with potentially partially written on-disk state.
+   */
+  public boolean canRestore();
```

Comment:
From a general resource perspective, would it be reasonable for a resource management system to assign persistent resources to this task if canRestore is true? What sort of rules are there about how long those persistent resources should be kept in the absence of a task using them?